be/src/column/array_column.cpp

```diff
@@ -512,4 +512,28 @@ StatusOr<ColumnPtr> ArrayColumn::downgrade() {
     return downgrade_helper_func(&_elements);
 }
 
+bool ArrayColumn::empty_null_array(NullColumnPtr null_map) {
+    DCHECK(null_map->size() == this->size());
+    bool need_empty = false;
+    auto size = this->size();
+    for (auto i = 0; i < size && !need_empty; ++i) {
+        if (null_map->get_data()[i] && _offsets->get_data()[i + 1] != _offsets->get_data()[i]) {
```

Comment:
here, we can use a simd algorithm.
1.  collect the null bits in a int32_t a, if it is not zero, then:
2. _offsets[i+1..i+9] - _offsets[i..i+8], collect the nonzero into a int32_t b
3. if a & b is nozero, then need_empty=true.

we can add such a algorithm in simd.h

Reply:
good suggestioin, but it is early to do optimization, let it as a TODO.